src/EnergyPlus/UserDefinedComponents.hh

```diff
@@ -69,10 +69,12 @@ namespace UserDefinedComponents {
     struct PlantConnectionStruct
     {
         // Members
-        int ErlInitProgramMngr; // points to an EMS:ProgramManager to run for setup and sizing
-        int ErlSimProgramMngr;  // points to an EMS:ProgramManager to run only when this connection is called
-        int simPluginLocation;  // If Python Plugins are used to simulate this, this defines the location in the plugin structure
-        int initPluginLocation; // If Python Plugins are used to init this, this defines the location in the plugin structure
+        int ErlInitProgramMngr;     // points to an EMS:ProgramManager to run for setup and sizing
+        int ErlSimProgramMngr;      // points to an EMS:ProgramManager to run only when this connection is called
+        int simPluginLocation;      // If Python Plugins are used to simulate this, this defines the location in the plugin structure
+        int initPluginLocation;     // If Python Plugins are used to init this, this defines the location in the plugin structure
+        int simCallbackIndex = -1;  // If API Callbacks are used to simulate this, this defines the index in the user-defined callbacks vector
+        int initCallbackIndex = -1; // If API Callbacks are used to init this, this defines the index in the user-defined callbacks vector
```

Comment:
Almost nerd sniped myself into refactoring these structs into a hierarchy, but for the sake of just getting this done, I held off..... for now.